roseau/load_flow/models/buses.py

```diff
@@ -39,15 +47,40 @@ def __init__(
             n:
                 Number of ports ie number of phases.
 
+            geometry:
+                The geometry of the bus.
+
             potentials:
                 List of initial potentials of each phase.
 
-            geometry:
-                The geometry of the bus.
+            ground:
+                The ground of the bus.
+
+            source_voltages:
+                The source voltages of the bus. If provided, it is used to fix the bus voltages.
+                In other terms, the bus becomes a slack bus which by definition a bus with a known
+                voltage. A non-slack bus must not set a source_voltages.
         """
         super().__init__(**kwargs)
         self.id = id
         self.n = n
+        if source_voltages is None:
+            # The bus is a normal bus (non slack): PV bus or PQ bus.
+            self.type = "bus" if n < 4 else "bus_neutral"
+        else:
+            # The bus is a slack bus: The voltage is known and fixed.
+            self.type = "slack"
```

Comment:
I don't really like the term "slack". As far as I understand electrical engineers, they use this word to define the main source voltage of a network (generally the largest), and there is only one slack bus per network. Here, we can have as many of this kind of buses in the network. It is also related to the method they use to solve the load flow problem (with a backward-forward, they take the root of the tree, ...) and if they use the bus of the largest generator that is often to ensure a "better convergence"/"better stability", and so on. Nevertheless, no electrical engineers in the G2ELab was able to explain me this in detail…

Let's give to this type a name to describe what it is: "voltage_source_bus".

Another question: There are two ways to define a voltage source:
1. We can define the phase-to-neutral voltages if there are 4 wires connected to this source ("Wye configuration")
2. We can define phase-to-phase voltages if there are only three wires connected to this source ("Delta configuration")
We should consider adding this option (maybe later) and therefore, I would recommend "voltage_source_bus" and "voltage_source_bus_neutral".